mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/dirtiness/ReindexOnUpdate.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.entity.pojo.dirtiness;
+
+/**
+ * Defines the various strategies for handling updates to a value in an entity.
+ * <p>
+ * A "value" here means either an entity property or something extracted from that property
+ * using a {@link org.hibernate.search.v6poc.entity.pojo.extractor.ContainerValueExtractor}.
+ */
+public enum ReindexOnUpdate {
+
+	/**
+	 * Track updates to the values used in the indexing process
+	 * (or to any nested values that may be used in the indexing process)
+	 * and trigger reindexing whenever an update is detected.
+	 * <p>
+	 * Note that nested values that are assigned {@link #NO} will still be ignored.
+	 */
+	USED,
```

Comment:
Can't say I'm excited about the name. Why `USED` instead of `YES`?

Reply:
The rationale was that if it's unclear, at least it's not misleading, and you're more likely to actually check the javadoc, as opposed to assuming some (wrong) behavior.

Something like that:

```
@Indexed
@MyTypeBridge
public class MyEntity {
  @IndexingDependency(reindexOnChange = ReindexOnChange.YES)
  private String something;
}
```

... could very well be assumed to trigger reindexing whenever `something` changes, because it is used (without being declared) in the type bridge.

While this:

```
@Indexed
@MyTypeBridge
public class MyEntity {
  @IndexingDependency(reindexOnChange = ReindexOnChange.USED)
  private String something;
}
```

... would at least give me pause to wonder what "used" means exactly (which the javadoc could explain to me).


At least I could understand it that way. Now if you think it's far fetched, ok, let's rename it to `YES`.

Reply:
I think `ENABLE` / `DISABLE` would be a good compromise. It doesn't say it will do it, just that the feature is enabled.

We can probably discuss this point when we will talk about all the API we expose.